tests/correctness/ntt-pipeline/ntt-16.expect

```diff
@@ -1,5 +1,5 @@
 {
-  "cycles": 252,
+  "cycles": 249,
```

Comment:
What caused this change?

Reply:
Pretty sure the only thing  that changed here is `done` holes guarded by a single port such as,
`groupX[done] = some_reg.done ? 1'd1;` was changed to `groupX[done] = some_reg.done;`.

Reply:
Can you make sure that this happens on master if you make this change manually? I'll be surprised if changing that assignment would change the cycle count.

Reply:
I'm not following. What do you mean?

Reply:
Can you isolate what actually caused this change? I.e., can you manually make the change to the FuTIL program on master and see if you see the number of cycles change?

My guess would be that `infer-static-timing` is can now analyze the program and generates slightly better schedules.

Reply:
I tried a fresh branch, updated to latest, `cargo rebuild`, old cycle counts. To confirm, I ran with old version of `gen-ntt-pipeline.py` on this branch and indeed cycle counts go to old counts.

Reply:
I guess I wasn't clear: I want us to isolate the change to the Calyx program, not the frontend. Can you demonstrate to me that the _only_ thing that causes the changes is changing:
```
groupX[done] = some_reg.done ? 1'd1
```
to
```
groupX[done] = some_reg.done;
```

I'm suspicious because I don't know if this is caused by a verilator issue or something.

Reply:
By changing a single group in the `precursor` stage from:
```
    group precursor_0 {
      r0.in = A0.out;
      r0.write_en = 1'd1;
      precursor_0[done] = r0.done; // CHANGE HERE
    }
```

to
```
    group precursor_0 {
      r0.in = A0.out;
      r0.write_en = 1'd1;
      precursor_0[done] = r0.done ? 1'd1; // CHANGE HERE
    }
```

I get the cycle difference.

`precursor` groups run in parallel before each stage except the first, so:
```
par { 
  precursor0 ; 
  ... ; 
  precursor(n-1); 
}
```
 runs `log2(n) - 1` times.

Reply:
Okay, it's probably `infer-static-timing` running into the issue in #310. Consider this resolved.